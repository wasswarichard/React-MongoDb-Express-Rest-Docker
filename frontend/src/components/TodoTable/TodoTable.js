import React, {useEffect, useState} from "react";
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';
import {store} from "../../state/store/store";

const columns = [
    {id: 'createdAt', label: "Created Date", minWidth: 170, format: value => new Date (value).toLocaleDateString()},
    {id: 'dueDate', label: "Due Date", minWidth: 170, format: value => new Date (value).toLocaleDateString()},
    {id: 'status', label: "Status", minWidth: 170, format: value => value.toLowerCase()},
    {id: 'text', label: "Text", minWidth: 170, format: value => value},
    {id: 'todoId', label: "Todo ID", minWidth: 170, format: value => value},
    {id: 'updatedAt', label: "UpDated Date", minWidth: 170, format: value => new Date (value).toLocaleDateString()},
]

const useStyles = makeStyles((theme) =>({
    head: {
        backgroundColor: "#1b4053",
        color: theme.palette.common.white,
    },
    root: {
        width: '100%',
    },
    container: {
        maxHeight: window.innerHeight,
    },
}));

export default function StickyHeadTable() {
    const classes = useStyles();
    const [page, setPage] = useState(0);
    const [rowsPerPage, setRowsPerPage] = useState(10);
    const [todos, setTodos] = useState([]);

    useEffect(() => {
        setTodos(store.getState().todos)
    }, [store.getState().todos])

    const handleChangePage = (event, newPage) => {
        setPage(newPage);
    };

    const handleChangeRowsPerPage = (event) => {
        setRowsPerPage(+event.target.value);
        setPage(0);
    };

    return (
        <div className="main-content">
            <main>
                <h2>Todos</h2>
                <Paper className={classes.root}>
                    <TableContainer className={classes.container}>
                        <Table stickyHeader aria-label="sticky table">
                            <TableHead>
                                <TableRow>
                                    {columns.map((column) => (
                                        <TableCell
                                            key={column.id}
                                            className={classes.head}
                                            // align={column.align}
                                            style={{ minWidth: column.minWidth }}
                                        >
                                            {column.label}
                                        </TableCell>
                                    ))}
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {todos.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((todo) => {
                                    return (
                                        <TableRow hover role="checkbox" tabIndex={-1} key={todo._id}>
                                            {columns.map((column) => {
                                                const value = todo[column.id];
                                                return (
                                                    <TableCell key={column.id} align={column.align}>
                                                        {column.format(value)}
                                                    </TableCell>
                                                );
                                            })}
                                        </TableRow>
                                    );
                                })}
                            </TableBody>
                        </Table>
                    </TableContainer>
                    <TablePagination
                        rowsPerPageOptions={[10, 25, 100]}
                        component="div"
                        count={todos.length}
                        rowsPerPage={rowsPerPage}
                        page={page}
                        onChangePage={handleChangePage}
                        onChangeRowsPerPage={handleChangeRowsPerPage}
                    />
                </Paper>
            </main>
        </div>
    );
}