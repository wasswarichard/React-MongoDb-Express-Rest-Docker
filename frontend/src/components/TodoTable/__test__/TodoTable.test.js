import React from "react";
import ReactDOM from "react-dom";
import TodoTable from "../TodoTable";
import { render, cleanup} from "@testing-library/react";
import TestRenderer from 'react-test-renderer';

afterEach(cleanup)
it('should render without crashing', function () {
    const div = document.createElement("div")
    ReactDOM.render(<TodoTable></TodoTable>, div)
});

it('should match snapshot', function () {
    const tree = TestRenderer.create(<TodoTable></TodoTable>).toJSON();
    expect(tree).toMatchSnapshot();

});
