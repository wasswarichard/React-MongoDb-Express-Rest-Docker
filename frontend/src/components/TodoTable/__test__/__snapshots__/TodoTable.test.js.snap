// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="main-content"
>
  <main>
    <h2>
      Todos
    </h2>
    <div
      className="MuiPaper-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiTableContainer-root makeStyles-container-3"
      >
        <table
          aria-label="sticky table"
          className="MuiTable-root MuiTable-stickyHeader"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                Created Date
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                Due Date
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                Status
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                Text
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                Todo ID
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-head-1 MuiTableCell-stickyHeader"
                scope="col"
                style={
                  Object {
                    "minWidth": 170,
                  }
                }
              >
                UpDated Date
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Previous page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Next page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
